### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by amashutikov**
[DEMO LINK](https://amashutikov.github.io/react_people-table-basics/)

### CODE REVIEW

File: src/pages/PepoplePage.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+          {isNoPeopleOnServer
+            ? (
+              <p data-cy="noPeopleMessage">
+                There are no people on the server
+              </p>
+            )
+            : (peopleFromServer?.length !== 0 && (
+              <PeopleTable peopleFromServer={peopleFromServer} />
+            ))}
```
Comment (dinaminahi) [inline src/pages/PepoplePage.tsx:56]:
```suggestion
          !peopleFromServer?.length
            ?  <p data-cy="noPeopleMessage">
                There are no people on the server
              </p>
            : <PeopleTable peopleFromServer={peopleFromServer} />
            }
```

File: src/components/TablePersonParents.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                  />
+                </td>
+              )
+              : <td>{person.motherName}</td>}
+          </>
+        )
+        : (
+          <td>-</td>
+        )}
+
```
Comment (dinaminahi) [inline src/components/TablePersonParents.tsx:36]: Soo confusing, create some variables, and check if you need so many react fragments

**Review by dinaminahi — CHANGES_REQUESTED** [review]
Good job! 👍 
But let's do some improvements

File: src/components/PeopleTable.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import { useParams } from 'react-router-dom';
+import { Person } from '../types';
+import { PersonLink } from './PersonLink';
+import { TablePersonParents } from './TablePersonParents';
+
+type Props = {
+  peopleFromServer: Person[];
+};
+
+const tableHeads: string[] = [
```
Comment (silchyn) [inline src/components/PeopleTable.tsx:12]:
```suggestion
const columns: string[] = [
```

File: src/components/PeopleTable.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  'Born',
+  'Died',
+  'Mother',
+  'Father',
+];
+
+export const PeopleTable: React.FC<Props> = ({ peopleFromServer }) => {
+  const { personSlug = '' } = useParams();
+
+  const findParent = useCallback((motherName: string) => {
```
Comment (silchyn) [inline src/components/PeopleTable.tsx:24]: Is it only mother?

File: src/components/PersonLink.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import cn from 'classnames';
+import { Person } from '../types';
+
+type Props = {
+  person: Person | string;
+};
+
+export const PersonLink: React.FC<Props> = ({ person }) => {
+  return (
```
Comment (silchyn) [inline src/components/PersonLink.tsx:9]:
```suggestion
export const PersonLink: React.FC<Props> = ({ person }) => (
```

File: src/components/TablePersonParents.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  return (
+    <>
+      {personMother
+        ? (
+          <td>
+            <PersonLink person={personMother} />
+          </td>
+        )
+        : <td>-</td>}
```
Comment (silchyn) [inline src/components/TablePersonParents.tsx:33]:
```suggestion
      <td>
        {personMother
          ? <PersonLink person={personMother} />
          : '-'}
      </td>
```

File: src/pages/PepoplePage.tsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
+import { useEffect, useState } from 'react';
+import { Loader } from '../components/Loader';
+import { getPeople } from '../api';
+import { Person } from '../types';
+
+import { PeopleTable } from '../components/PeopleTable';
+
+export const PeoplePage: React.FC = () => {
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isDownloading, setIsDownloading] = useState(false);
```
Comment (silchyn) [inline src/pages/PepoplePage.tsx:10]: What is downloading?

File: src/pages/PepoplePage.tsx [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+  useEffect(() => {
+    setIsDownloading(true);
+
+    getPeople()
+      .then(result => {
+        if (result.length === 0) {
+          setPeopleFromServer([]);
+
+          return;
+        }
```
Comment (silchyn) [inline src/pages/PepoplePage.tsx:25]: What's the point?

File: src/pages/PepoplePage.tsx [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+      });
+  }, []);
+
+  return (
+    <div className="container">
+      <h1 className="title">People Page</h1>
+
+      <div className="block">
+        <div className="box table-container">
+          {(isDownloading && !peopleFromServer) && <Loader />}
```
Comment (silchyn) [inline src/pages/PepoplePage.tsx:41]: Put it in a separate variable

**Review by silchyn — CHANGES_REQUESTED** [review]

**Pull request closed without merging**